### PR TITLE
api: Add sync_fd semaphore test

### PR DESCRIPTION
**Pull request by lakshmih**
Add subtest to verify correct behavior of
SYNC_FD semaphores.  Signal and wait multiple
semaphores across different OpenCL contexts.

### CODE REVIEW

**Review by paulfradgley — COMMENTED** [review]
Looks good to me.

**Comment by bashbaug** [comment]
It would be nice to have somebody from ARM review this - @kpet?

**Comment by JohnKesapidesARM** [comment]
I've been asked by @kpet to have a look on behalf of ARM. I'll try and get this done today

**Review by JohnKesapidesARM — APPROVED** [review]
Looks good to me.

File: test_conformance/api/test_semaphores.cpp [hunk 1]
```diff
@@ -1058,89 +1057,209 @@ int test_semaphores_invalid_command(cl_device_id deviceID, cl_context context,
...
+// Test that handles with value -1 import as signaled
+int test_semaphores_import_signaled(cl_device_id deviceID, cl_context context,
+                                    cl_command_queue defaultQueue,
+                                    int num_elements)
+{
+    cl_int err;
+
+    if (!is_extension_available(deviceID, "cl_khr_semaphore"))
+    {
+        log_info("cl_khr_semaphore is not supported on this platoform. "
```
Comment (OleStrohm) [inline test_conformance/api/test_semaphores.cpp:1212]:
```suggestion
        log_info("cl_khr_semaphore is not supported on this platform. "
```

File: test_conformance/api/test_semaphores.cpp [hunk 2]
```diff
@@ -1058,89 +1057,209 @@ int test_semaphores_invalid_command(cl_device_id deviceID, cl_context context,
...
+                 "Skipping test.\n");
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    if (!is_extension_available(deviceID, "cl_khr_external_semaphore_sync_fd"))
+    {
+        log_info("cl_khr_external_semaphore_sync_fd is not supported on this "
+                 "platoform. Skipping test.\n");
```
Comment (OleStrohm) [inline test_conformance/api/test_semaphores.cpp:1220]:
```suggestion
                 "platform. Skipping test.\n");
```

File: test_conformance/api/test_semaphores.cpp [hunk 3]
```diff
@@ -1058,89 +1057,209 @@ int test_semaphores_invalid_command(cl_device_id deviceID, cl_context context,
...
+        clCreateCommandQueue(context, deviceID, 0, &err);
+    test_error(err, "Could not create command queue");
+
+    // Import signaled sync_fd handle
+    int handle = -1;
+
+    cl_semaphore_properties_khr sema_props[] = {
+        static_cast<cl_semaphore_properties_khr>(CL_SEMAPHORE_TYPE_KHR),
+        static_cast<cl_semaphore_properties_khr>(CL_SEMAPHORE_TYPE_BINARY_KHR),
+        CL_SEMAPHORE_HANDLE_SYNC_FD_KHR,
```
Comment (OleStrohm) [inline test_conformance/api/test_semaphores.cpp:1241]:
```suggestion
        static_cast<cl_semaphore_properties_khr>(
            CL_SEMAPHORE_HANDLE_SYNC_FD_KHR),
```

File: test_conformance/api/test_semaphores.cpp [hunk 4]
```diff
@@ -1058,89 +1057,209 @@ int test_semaphores_invalid_command(cl_device_id deviceID, cl_context context,
...
+        test_error(err, "Could not extract semaphore handle");
+        test_assert_error(sizeof(handle) == handle_size, "Invalid handle size");
+        test_assert_error(handle >= 0, "Invalid handle");
+
+        // Create semaphore from sync fd
+        cl_semaphore_properties_khr sema_2_props[] = {
+            static_cast<cl_semaphore_properties_khr>(CL_SEMAPHORE_TYPE_KHR),
+            static_cast<cl_semaphore_properties_khr>(
+                CL_SEMAPHORE_TYPE_BINARY_KHR),
+            CL_SEMAPHORE_HANDLE_SYNC_FD_KHR,
```
Comment (OleStrohm) [inline test_conformance/api/test_semaphores.cpp:1147]:
```suggestion
            static_cast<cl_semaphore_properties_khr>(
                CL_SEMAPHORE_HANDLE_SYNC_FD_KHR),
```

**Comment by lakshmih** [comment]
This PR has been superseded by #1747 
We plan to close this PR soon.

**Pull request closed without merging**